# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsapi/TransceiverImpl.cpp
# Commit: 0991ebfeb365
# Full Hash: 0991ebfeb3657ca4e5c379c5211f6f4d1e29968d
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 1
# Description:
#   Bug 1654112 - Make RTCDTMFSender signal DTMF tones through a MediaEvent. r=ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D112859
# ==============================================================================

diff -r bd3db9010131 -r 0991ebfeb365 dom/media/webrtc/jsapi/TransceiverImpl.cpp
--- a/dom/media/webrtc/jsapi/TransceiverImpl.cpp	Sun Apr 18 21:47:29 2021 +0200
+++ b/dom/media/webrtc/jsapi/TransceiverImpl.cpp	Sun Apr 18 21:06:49 2021 +0200
@@ -79,8 +79,7 @@
                                  aMainThread, aStsThread, mConduit, this);
 
   if (!IsVideo()) {
-    mDtmf = new RTCDTMFSender(
-        aWindow, this, static_cast<AudioSessionConduit*>(mConduit.get()));
+    mDtmf = new RTCDTMFSender(aWindow, this);
   }
 
   mTransmitPipeline =
@@ -727,7 +726,7 @@
             [](const auto& a, const auto& b) { return a.mFreq < b.mFreq; });
       }
       MOZ_ASSERT(dtmfIterator != dtmfConfigs.end());
-      conduit->SetDtmfPayloadType(dtmfIterator->mType, dtmfIterator->mFreq);
+      mDtmf->SetPayloadType(dtmfIterator->mType, dtmfIterator->mFreq);
     }
 
     auto error = conduit->ConfigureSendMediaCodec(configs[0]);
